safe-global/safe-react
# fix: Allow estimation for spending limit txs

## What it solves
Resolves #3209 

## How this PR fixes it
Displays estimation data and allows editing when executing a spending limit transaction.

Some refactoring to `TxModalWrapper` was necessary in order to make this work with different contracts

- Split up `useEstimateTransactionGas` into `useEstimateGasLimit` and `useExecutionStatus`
- Compute `gasCostFormatted` outside of the custom hook since it now relies on multiple custom hooks
- Removed some redundant conditions and logic from `TxEstimatedFeesDetail`

## Open issues

#### Pending state
Displaying a pending state only works for transactions that are proposed to the backend and have a `txId`. Spending limit transactions are executed directly on the spending limit contract and are not proposed to the backend, have no `txId` and don't show up in the Queue.

## How to test it
Creating, signing and executing transactions needs to be tested for different safe versions

1. Open the Safe app
2. Create a new spending limit tx
6. Observe that `safeNonce` doesn't exist
3. Observe `gasLimit`, `gasPrice`, `maxPrioFee` being displayed
4. Change any of these values
5. Observe that the estimated cost updates 
6. Submit the Tx
7. Observe that there is a sign notification popping up
8. Reject the Tx
9. Observe that there is an error notification popping up

## Screenshots
<img width="545" alt="Screenshot 2022-05-18 at 10 32 55" src="https://user-images.githubusercontent.com/5880855/168994960-d115ca52-e8a6-47bb-8708-7776a90337e4.png">
<img width="469" alt="Screenshot 2022-05-18 at 10 37 24" src="https://user-images.githubusercontent.com/5880855/168995882-a58f2487-d68c-418d-837f-433338a88f39.png">
